internal/service/rds/cluster.go

```diff
@@ -1149,8 +1149,11 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, meta int
 			input.DBClusterParameterGroupName = aws.String(d.Get("db_cluster_parameter_group_name").(string))
 		}
 
-		if d.HasChange("db_instance_parameter_group_name") {
-			input.DBInstanceParameterGroupName = aws.String(d.Get("db_instance_parameter_group_name").(string))
+		// DB instance parameter group name is not currently returned from the
+		// DescribeDBClusters API. This means there is no drift detection, so when
+		// set, the configured attribute should always be sent on modify.
+		if v, ok := d.GetOk("db_instance_parameter_group_name"); ok || d.HasChange("db_instance_parameter_group_name") {
```

Comment:
I believe `d.GetOk("db_instance_parameter_group_name")` would always get the configured value whether there is a change or not. `d.HasChange()` might not be needed here.

This should work similar to how `d.Get("apply_immediately)` works in an update. It is always passed into input

Reply:
Yeah, I wasn't positive here - I kept `d.HasChange()` to preserve backward compatibility with the previous behavior.

Based on the old implementation, I assumed an empty input was required to remove an existing parameter group if the configuration removes it (while omitting the input would leave it unchanged). If omitting the value causes the same behavior as an empty input I think we can safely remove though.